### PR TITLE
[SPARK-15053][BUILD] Fix Java Lint errors on Hive-Thriftserver module

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
## What changes were proposed in this pull request?

This issue fixes or hides 181 Java linter errors introduced by SPARK-14987 which copied hive service code from Hive. We had better clean up these errors before releasing Spark 2.0.
- Fix UnusedImports (15 lines), RedundantModifier (14 lines), SeparatorWrap (9 lines), MethodParamPad (6 lines), FileTabCharacter (5 lines), ArrayTypeStyle (3 lines), ModifierOrder (3 lines), RedundantImport (1 line), CommentsIndentation (1 line), UpperEll (1 line), FallThrough (1 line), OneStatementPerLine (1 line), NewlineAtEndOfFile (1 line) errors.
- Ignore `LineLength` errors under `hive/service/*` (118 lines).
- Ignore `MethodName` error in `PasswdAuthenticationProvider.java` (1 line).
- Ignore `NoFinalizer` error in `ThreadWithGarbageCleanup.java` (1 line).
## How was this patch tested?

After passing Jenkins building, run `dev/lint-java` manually.

``` bash
$ dev/lint-java 
Checkstyle checks passed.
```

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
PySpark error is irrelevant to this PR. I'll rebase this to retrigger Jenkins.

File: dev/checkstyle-suppressions.xml [hunk 1]
```diff
@@ -36,4 +36,10 @@
               files="src/test/java/org/apache/spark/sql/hive/test/Complex.java"/>
     <suppress checks="LineLength"
               files="src/main/java/org/apache/spark/examples/JavaLogQuery.java"/>
+    <suppress checks="LineLength"
+              files="src/main/java/org/apache/hive/service/*"/>
+    <suppress checks="MethodName"
+              files="src/main/java/org/apache/hive/service/auth/PasswdAuthenticationProvider.java"/>
```
Comment (hvanhovell) [inline dev/checkstyle-suppressions.xml]: How hard is it to fix this one?

**Comment by srowen** [comment]
I agree with all the changes, though some look like they wouldn't have actually been lint errors? I assume we can't fix the `PasswdAuthenticationProvider` file because it's cloned from Hive.

**Comment by dongjoon-hyun** [comment]
Thank you for review, @hvanhovell and @srowen .
For `PasswdAuthenticationProvider`, I didn't change that due to the same reason mentioned by @srowen .

**Comment by srowen** [comment]
Merged to master/2.0

**Comment by dongjoon-hyun** [comment]
Thank you, @srowen .

**Pull request closed without merging**